rfcs/0000-c-unwind-abi.md

```diff
@@ -0,0 +1,273 @@
+- Feature Name: `"C unwind" ABI`
+- Start Date: 2019-04-03
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Introduces a new ABI string, `"C unwind"`, to enable unwinding from other
+languages (such as C++) into Rust frames and from Rust into other languages.
+
+Additionally, defines the behavior for a limited number of previously-undefined
+cases when an unwind operation reaches a Rust function boundary with a
+non-`"Rust"`, non-`"C"` ABI.
+
+This RFC does not define the behavior of `catch_unwind` in a Rust frame being
+unwound by a foreign exception.
+
+# Motivation
+[motivation]: #motivation
+
+There are some Rust projects that need cross-language unwinding to provide
+their desired functionality. One major example is WASM interpreters, including
+the Lucet and Wasmer projects.
+
+There are also existing Rust crates (notably, wrappers around the `libpng` and
+`libjpeg` C libraries) that rely on compatibility with the native exception
+handling mechanism in GCC, LLVM, and MSVC.
+
+Additionally, there are libraries such as `rlua` that rely on `longjmp` across
+Rust frames; on Windows, `longjmp` is implemented via unwinding.
+
+The desire for this feature has been previously discussed on other RFCs,
+including #2699 and #2753.
```

Comment:
```suggestion
including [#2699](https://github.com/rust-lang/rfcs/pull/2699) and [#2753](https://github.com/rust-lang/rfcs/pull/2573).
```

links ftw

Reply:
Hm, I thought that wasn't necessary since this will be posted in the RFC repository, so the links will work automatically when rendered by GitHub in that repo. I suppose it's better to be explicit, though, especially since the current version is in this repo.